coorx/coordinates.py

```diff
@@ -59,60 +64,93 @@ def __len__(self):
 
     def __getitem__(self, index):
         return self.coordinates[index]
-    
+
     def __iter__(self):
-        for x in self.coordinates:
-            yield x
+        yield from self.coordinates
 
     def _check_operand(self, a):
-        assert isinstance(a, PointArray)
-        assert a.system is self.system
+        if not isinstance(a, PointArray):
+            raise TypeError(f"Operand must be a PointArray (received {type(a)})")
+        if a.system is not self.system:
+            raise ValueError(f"Operand system {a.system} does not match this PointArray's system {self.system}")
 
     def __add__(self, b):
-        self._check_operand(b)
-        return self.coordinates + b.coordinates
-
-    def __sub__(self, b):
-        self._check_operand(b)
-        return self.coordinates - b.coordinates
-
-    def mapped_through(self, cs_list):
+        if isinstance(b, (Vector, VectorArray)):
+            self._check_vector_operand(b)
+            new_coords = self.coordinates + b.displacement
+            if isinstance(self, Point) and isinstance(b, Vector):
+                return Point(new_coords, system=self.system)
+            else:
+                return PointArray(new_coords, system=self.system)
+        else:
+            self._check_point_operand(b)
+            # This case should likely not return raw coordinates, but maybe another PointArray?
+            # For now, keeping original behavior for PointArray + PointArray.
+            # Consider if PointArray + PointArray should be allowed or raise TypeError.
+            return self.coordinates + b.coordinates
+
+    def __sub__(self, b: PointArray) -> VectorArray:
+        self._check_point_operand(b)
+        if isinstance(self, Point) and isinstance(b, Point):
```

Comment:
Should all these isinstance(Point) just be written as overrides on the Point class?